.github/workflows/test.yml

```
@@ -143,31 +143,90 @@ jobs:
+            fi
+          done
+          echo "sonar.go.coverage.reportPaths=$(IFS=,; echo "${COVERAGE_PATHS[*]}")" >> sonar-project.properties
+          ls -la "${COVERAGE_PATHS[@]}" report.json report*.json 2>/dev/null || true
+      - name: Prepare test report for SonarCloud
+        run: |
+          if [ -f report_unit.json ]; then
+            cp report_unit.json report.json
+          fi
+      - name: Set the PR properties in SonarCloud
+        if: github.event_name == 'pull_request'
+        run: |
```

actionlint (1.7.12)

[error] "github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)